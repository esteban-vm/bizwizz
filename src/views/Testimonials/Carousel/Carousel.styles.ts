import clsx from 'clsx'

const styles = {
  wrapper: clsx('flex flex-col overflow-hidden'),
  radio: clsx(
    'hidden carousel-check:bg-primary',
    'carousel-slide-2:translate-x-[calc(-100%/3)] carousel-slide-3:translate-x-[calc(-100%*2/3)]'
  ),
  label: clsx('m-[1.6rem] block h-5 w-5 cursor-pointer rounded-full border-2 border-solid border-primary bg-white'),
  label_container: clsx('order-1 flex justify-center'),
  slider: clsx('flex w-[300%] transition-all duration-500'),
}

export default styles
