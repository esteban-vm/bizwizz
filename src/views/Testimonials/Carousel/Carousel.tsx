'use client'
/* eslint-disable jsx-a11y/label-has-associated-control */
import type { FC, PropsWithChildren } from '@/types'
import { useRef, useEffect } from 'react'
import { Icon } from '@/components'
import styles from './Carousel.styles'

const Carousel: FC<PropsWithChildren> = ({ children }) => {
  const slide1Ref = useRef<HTMLInputElement>(null)
  const slide2Ref = useRef<HTMLInputElement>(null)
  const slide3Ref = useRef<HTMLInputElement>(null)

  useEffect(() => {
    if (slide1Ref.current) {
      slide1Ref.current.checked = true
    }
  }, [])

  const leftClick = () => {
    if (slide1Ref.current && slide2Ref.current && slide3Ref.current) {
      if (slide1Ref.current.checked) slide3Ref.current.checked = true
      else if (slide2Ref.current.checked) slide1Ref.current.checked = true
      else if (slide3Ref.current.checked) slide2Ref.current.checked = true
    }
  }

  const rightClick = () => {
    if (slide1Ref.current && slide2Ref.current && slide3Ref.current) {
      if (slide1Ref.current.checked) slide2Ref.current.checked = true
      else if (slide2Ref.current.checked) slide3Ref.current.checked = true
      else if (slide3Ref.current.checked) slide1Ref.current.checked = true
    }
  }

  return (
    <div className={styles.outer_wrapper}>
      <Icon className={styles.icon} name='chevron-left' onClick={leftClick} />
      <div className={styles.inner_wrapper}>
        <input ref={slide1Ref} className={styles.radio} id='slide_1' name='slide' type='radio' />
        <input ref={slide2Ref} className={styles.radio} id='slide_2' name='slide' type='radio' />
        <input ref={slide3Ref} className={styles.radio} id='slide_3' name='slide' type='radio' />
        <div className={styles.label_container}>
          <label className={styles.label} htmlFor='slide_1' />
          <label className={styles.label} htmlFor='slide_2' />
          <label className={styles.label} htmlFor='slide_3' />
        </div>
        <div className={styles.slider}>{children}</div>
      </div>
      <Icon className={styles.icon} name='chevron-right' onClick={rightClick} />
    </div>
  )
}

export default Carousel
