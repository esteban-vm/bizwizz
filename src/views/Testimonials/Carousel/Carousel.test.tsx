import { render, cleanup } from '@/tests'
import Carousel from './Carousel'

describe('Carousel:', () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    void ({ asFragment } = render(<Carousel />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })
})
