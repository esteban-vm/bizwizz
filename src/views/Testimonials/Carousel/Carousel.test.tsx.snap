// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Carousel: > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col overflow-hidden"
  >
    <input
      checked=""
      class="hidden carousel-check:bg-primary carousel-slide-2:translate-x-[calc(-100%/3)] carousel-slide-3:translate-x-[calc(-100%*2/3)]"
      id="slide_1"
      name="slide"
      type="radio"
      value="1"
    />
    <input
      class="hidden carousel-check:bg-primary carousel-slide-2:translate-x-[calc(-100%/3)] carousel-slide-3:translate-x-[calc(-100%*2/3)]"
      id="slide_2"
      name="slide"
      type="radio"
      value="2"
    />
    <input
      class="hidden carousel-check:bg-primary carousel-slide-2:translate-x-[calc(-100%/3)] carousel-slide-3:translate-x-[calc(-100%*2/3)]"
      id="slide_3"
      name="slide"
      type="radio"
      value="3"
    />
    <div
      class="order-1 flex justify-center"
    >
      <label
        class="m-[1.6rem] block h-5 w-5 cursor-pointer rounded-full border-2 border-solid border-primary bg-white"
        for="slide_1"
      />
      <label
        class="m-[1.6rem] block h-5 w-5 cursor-pointer rounded-full border-2 border-solid border-primary bg-white"
        for="slide_2"
      />
      <label
        class="m-[1.6rem] block h-5 w-5 cursor-pointer rounded-full border-2 border-solid border-primary bg-white"
        for="slide_3"
      />
    </div>
    <div
      class="flex w-[300%] transition-all duration-500"
    />
  </div>
</DocumentFragment>
`;
